app/views.py

```diff
@@ -11,109 +20,82 @@ def home():
 
     return render_template("index.html")
 
+
 @app.route("/show_users")
 def users():
-    conn = database_context.connect()
+    
+    users = UsersRepository.fetchUsers()
 
-    c = conn.cursor()
+    info = []
 
-    c.execute("SELECT * FROM user")
+    for user in users:
+        info.append(' '.join(str(info) for info in user.getUserDetails()))
 
-    users = c.fetchall()
+    return '<br>'.join(info)
 
-    return '<br>'.join(' '.join(str(value) for value in list(user)) for user in list(users))
 
 @app.route("/decks")
 def decks():
-    conn = database_context.connect()
 
-    cursor = conn.cursor()
+    decks = DecksRepository.fetch_decks()
 
-    query = "SELECT * FROM deck;"
+    return render_template("show_decks.html", decks = decks)
 
-    cursor.execute(query)
 
-    results = cursor.fetchall()
-
-    return render_template("show_decks.html", decks = results)
-# TODO deck/id/cards
 @app.route("/deck/<deck_id>")
 def display_cards(deck_id):
 
-    conn= database_context.connect()
-
-    cursor = conn.cursor()
-
-    querry = f"SELECT * FROM card WHERE deck_id == {deck_id};"
+    deck = DecksRepository.fech_deck_by_id(deck_id)
 
-    cursor.execute(querry)
+    return render_template("show_cards.html", deck = deck)
 
-    cards = cursor.fetchall()
-
-    return render_template("show_cards.html", cards = cards, deck_id = deck_id)
 
 @app.route("/deck/<deck_id>/card/<card_id>")
 def card_detail(deck_id, card_id):
     
-    conn = database_context.connect()
-
-    cursor = conn.cursor()
+    card = CardsRepository.returnCardById(card_id)
 
-    querry = f"SELECT word, translation FROM card WHERE deck_id == {deck_id} AND card_id == {card_id};"
+    return render_template("card_detail.html", card = card)
 
-    cursor.execute(querry)
-
-    card = cursor.fetchone()
-    
-    word = card[0]
-    translation = card[1]
-
-    return render_template("card_detail.html", word = word, translation = translation)
 
 @app.route("/create_deck", methods=["POST", "GET"])
 def create_deck():
 
-    
-    if request.method == "POST":
-        name = request.form['name']
-
-        conn = database_context.connect()
+    if not 'user' in session:
+        return redirect(url_for("login"))
 
-        sql_query = f"INSERT INTO deck (name) VALUES('{name}');"
+    if request.method == "POST":
 
-        cursor = conn.cursor()
+        name = request.form['name']
 
-        cursor.execute(sql_query)
+        user_id = json.loads(session['user'])['id']
 
-        conn.commit()
+        deck = DecksRepository.createDeck(user_id, name)
 
-        if cursor.rowcount == 1:
+        if DecksRepository.saveToDataBase(deck):
             return render_template("create_deck.html", success=True)
         else:
             return render_template("create_deck.html", success=False)
+
     else:
         return render_template("create_deck.html")
 
-
 @app.route("/deck/<id>/create_card", methods=["GET", "POST"])
 def create_card(id):
 
+    if not 'user' in session:
```

Comment:
just wondering can you write if 'user' not in session?

Reply:
I did it and had to correct :P

Reply:
check line 64, you are using it already, it should work here too if it works there.